### PR TITLE
Inductor cpp wrapper: support dynamic shapes

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97965

1. Fixed dynamic shapes support in cpp_wrapper
   - fixed the cpp codegen of `size()` and `stride()`
   - fixed the cpp codegen of `ShapeAsConstantBuffer`
   - changed to use `cexpr` instead of `pexpr` in the cpp codegen of the `sizevar`

2. Enabled dynamic shapes tests for cpp_wrapper

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -1669,6 +1669,25 @@ def fn(a, b):
...
+    def test_scalar_input(self):
+        def fn(x, y):
+            a = torch.div(x, y, rounding_mode="floor")
+            return a
+
+        self.common(fn, [torch.randint(5, (1, 8)), 5400])
+
+    def test_int_div(self):
+        if self.device == "cuda":
+            raise unittest.SkipTest("only support cpu int_div test")
```
Comment (jgong5) [inline test/inductor/test_torchinductor.py:1681]: If so, consider to move it to CPURepro?

**Review by jgong5 — APPROVED** [review]
LGTM. Suggest to move CPU-only test to the right place. See comment.

File: test/inductor/test_torchinductor.py (hunk 1 above)
Comment (chunyuan-w) [reply test/inductor/test_torchinductor.py]: Moved to `CPURepro`.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**